Tasks/ServiceFabricDeploy/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1

```diff
@@ -215,70 +216,84 @@ function Publish-UpgradedServiceFabricApplication
             throw $errMsg
         }
 
-        $reg = Get-ServiceFabricApplicationType -ApplicationTypeName $names.ApplicationTypeName | Where-Object  { $_.ApplicationTypeVersion -eq $names.ApplicationTypeVersion }
-        if ($reg)
-        {
-            Write-Host (Get-VstsLocString -Key SFSDK_UnregisteringExistingAppType -ArgumentList @($names.ApplicationTypeName, $names.ApplicationTypeVersion))
-            $reg | Unregister-ServiceFabricApplicationType -Force
+        $reg = Get-ServiceFabricApplicationType -ApplicationTypeName $names.ApplicationTypeName | Where-Object  { $_.ApplicationTypeVersion -eq $names.ApplicationTypeVersion }        
+        if ($reg){
+            $appIsInUse = $false
+            $apps = Get-ServiceFabricApplication -ApplicationTypeName $names.ApplicationTypeName
+            $apps | ForEach-Object {
+                if ($_.ApplicationTypeVersion -eq $names.ApplicationTypeVersion)
+                {
+                    $appIsInUse = $true
+                }
+            }
+            if(!$appIsInUse)
+            {
+                Write-Host (Get-VstsLocString -Key SFSDK_UnregisteringExistingAppType -ArgumentList @($names.ApplicationTypeName, $names.ApplicationTypeVersion))
+                $reg | Unregister-ServiceFabricApplicationType -Force
+                $ApplicationTypeAlreadyRegistered = $false        
+            }
```

Comment:
Please add a Write-Host in an else to note in the logs when we are skipping registration due to the existing type being in use.